README.md

```diff
@@ -12,6 +54,102 @@ public ResponseEntity<?> proxyRequest(@RequestHeader HttpHeaders httpHeaders,
 This methods is used in both REST and WSS flows.
 
 
+## Customizing DataApp
+
+If you need to modify dataApp, you can perform such modification in 2 places: consumer side or provider.
+
+### Consumer side modification
+
+Following class is used as entry point for consumer side:
+
+**it.eng.idsa.dataapp.web.rest.ProxyController**
+
+This class is the entry point of [proxy request](#proxyendpoint). Business logic for creating request is delegated to:
+
+**it.eng.idsa.dataapp.service.ProxyServiceImpl**
+
+This class wraps up logic for creating proper IDS Message, and proper request, based on the configuration (mixed, form or header) and sends request to consumer ECC.
+
+Once response is received, it will just log request. If you need to do something else with the response - check following method:
+
+**handleResponse(ResponseEntity<String> resp, MultipartMessage mm)**
+
+### Provider side modification
+
+For making modification when dataApp is in provider role, one of the following controllers can be used as starting point
+
+**it.eng.idsa.dataapp.web.rest.DataControllerBodyBinary**
+
+**it.eng.idsa.dataapp.web.rest.DataControllerBodyForm**
+
+**it.eng.idsa.dataapp.web.rest.DataControllerHttpHeader**
+
+Depending on the configuration, mixed, form or header.
+
+This class (controller) is an entry point in Provider part of the dataApp, and it will receive request from Provider ECC.
+Depending on the IDS Message received it will execute predefined logic, and that should not be changed. Only modification should be made when ArtifactRequestMessage is received - when creating response payload. Code of interest can be found in following method:
+
+**MessageUtil.it.eng.idsa.dataapp.util.createResponsePayload(Message requestHeader, String payload)**
+
+```
+else if (requestHeader instanceof ArtifactRequestMessage && isBigPayload(((ArtifactRequestMessage) requestHeader).getRequestedArtifact().toString())) {
+	return encodePayload == true ? encodePayload(BigPayload.BIG_PAYLOAD.getBytes()) : BigPayload.BIG_PAYLOAD;
+	}
+	return  encodePayload == true ? encodePayload(createResponsePayload().getBytes()) : createResponsePayload();
+			
+```
+Current example has 2 payloads, one small - json representing "John Doe" and other - big payload, that has several hundred lines of text. For your use case, you can simplify it and just handle *if (requestHeader instanceof ArtifactRequestMessage)* use case. 
+
+What is needed is to modify
+**private String createResponsePayload() ** method and provide logic that will fit your needs. You can add here part to read from some DB, call API, read file from filesystem, anything you need.
```

Comment:
the text is not bold, delete last space before **

Reply:
done